feature/foryou/src/main/java/com/google/samples/apps/nowinandroid/feature/foryou/ForYouViewModel.kt

```diff
@@ -174,7 +191,7 @@ class ForYouViewModel @Inject constructor(
                     }
 
                     if (topics.isEmpty() && authors.isEmpty()) {
-                        ForYouInterestsSelectionUiState.Loading
+                        ForYouInterestsSelectionUiState.LoadFailed
```

Comment:
Maybe the logic should be this (with no `if` above):
```suggestion
                        if (isSyncing) {
                            ForYouInterestsSelectionUiState.Loading
                        } else {
                            ForYouInterestsSelectionUiState.LoadFailed
                        }
```

Reply:
Applied and added a loading field to `ForYouInterestsSelectionUiState.WithInterestsSelection`. Would you model this differently? Is it weird to have two different kinds of loading states?

Reply:
Resolved in the comment above.